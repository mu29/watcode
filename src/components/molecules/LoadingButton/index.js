import React from 'react'
import PropTypes from 'prop-types'
import { Button, Loading } from 'components'

const LoadingButton = ({
  isLoading,
  loadingColor,
  loadingSize,
  children,
  ...props
}) => (
  <Button { ...props }>
    { isLoading ? <Loading color={ loadingColor } size={ loadingSize } /> : children }
  </Button>
)

LoadingButton.propTypes = {
  isLoading: PropTypes.bool.isRequired,
  loadingColor: PropTypes.string,
  loadingSize: PropTypes.number,
  children: PropTypes.node.isRequired,
}

LoadingButton.defaultProps = {
  loadingColor: 'white',
  loadingSize: 16,
}

export default LoadingButton
