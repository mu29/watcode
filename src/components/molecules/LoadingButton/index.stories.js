import React from 'react'
import { storiesOf } from '@storybook/react'
import LoadingButton from '.'

storiesOf('Molecules/LoadingButton', module)
  .add('기본', () => (
    <LoadingButton isLoading>
      로그인
    </LoadingButton>
  ))
