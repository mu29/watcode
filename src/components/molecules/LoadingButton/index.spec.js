import React from 'react'
import LoadingButton from '.'

const component = (props = {}) => withTheme(<LoadingButton { ...props }>확인</LoadingButton>)
const wrap = (props = {}) => shallow(component(props)).dive()

describe('<LoadingButton />', () => {
  it('정상적으로 렌더링된다', () => {
    const wrapper = render(component({ isLoading: true }))
    expect(wrapper).toMatchSnapshot()
  })

  it('로딩 중이 아니라면 자식 컴포넌트를 표시한다', () => {
    const wrapper = wrap({ isLoading: false })
    expect(wrapper).toMatchSnapshot()
  })
})
