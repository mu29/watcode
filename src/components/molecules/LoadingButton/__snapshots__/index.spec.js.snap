// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoadingButton /> 로딩 중이 아니라면 자식 컴포넌트를 표시한다 1`] = `
<span
  componentname="Button"
>
  확인
</span>
`;

exports[`<LoadingButton /> 정상적으로 렌더링된다 1`] = `
<span
  componentname="Button"
>
  <span
    componentname="Loading"
  />
</span>
`;
